official/vision/beta/projects/yt8m/dataloaders/video_input.py

```diff
@@ -0,0 +1,354 @@
+      feature_name: tf.io.FixedLenSequenceFeature([], dtype=tf.string)
+      for feature_name in self._feature_names
+    }
+
+  def decode(self, serialized_example):
+    """Parses a single tf.Example into image and label tensors."""
+
+    contexts, features = tf.io.parse_single_sequence_example(
+      serialized_example,
+      context_features=self._context_features,
+      sequence_features=self._sequence_features)
+
+    return contexts, features
+
+
+class Parser(parser.Parser):
+  """Parses a video and label dataset.
+    takes the decoded raw tensors dict
+    and parse them into a dictionary of tensors that can be consumed by the model.
+    It will be executed after decoder.
+  """
+
+  def __init__(self,
+               contexts,
+               features,
+               input_params: exp_cfg.DataConfig,
+               # image_key: str = IMAGE_KEY,
+               # label_key: str = LABEL_KEY,
+               max_quantized_value=2,
+               min_quantized_value=-2,
+               ):
+    self._num_classes = input_params.num_classes
+    # self._image_key = image_key
+    # self._label_key = label_key
+
+    self._segment_size = input_params.segment_size
+    self._segment_labels = input_params.segment_labels
+    self._feature_names = input_params.feature_names
+    self._feature_sizes = input_params.feature_sizes
+    self._max_frames = input_params.max_frames
+    self._max_quantized_value = max_quantized_value
+    self._min_quantized_value = min_quantized_value
+    self.features = features
+    self.contexts = contexts
+
+    # loads (potentially) different types of features and concatenates them
+    self.video_matrix, self.num_frames = _concat_features(self.features, self._feature_names, self._feature_sizes,
+                                                          self._max_frames, self._max_quantized_value,
+                                                          self._min_quantized_value)
+
+  def _parse_train_data(self):  # -> Tuple[Dict[str, tf.Tensor], tf.Tensor]
+    """Parses data for training."""
+    output_dict = _process_segment_and_label(self.video_matrix, self.num_frames, self.contexts, self._segment_labels,
+                                             self._segment_size, self._num_classes)
+
+    return output_dict  # batched
+
+  def _parse_eval_data(self):  # -> Tuple[Dict[str, tf.Tensor], tf.Tensor]
+    """Parses data for evaluation."""
+    output_dict = _process_segment_and_label(self.video_matrix, self.num_frames, self.contexts, self._segment_labels,
+                                             self._segment_size, self._num_classes)
+
+    return output_dict
+
+
+class PostBatchProcessor(object):
+  """Processes a video and label dataset which is batched."""
+
+  def __init__(self, input_params: exp_cfg.DataConfig):
+    self._segment_labels = input_params.segment_labels
+    self._is_training = input_params.is_training
+    self._num_test_clips = input_params.num_test_clips
```

Comment:
It is not instantly clear to me what num_test_clips is here for. Please comment here with more details.

Reply:
The num_test_clips option is probably inherit from the video_input.py. This option is for using multiple segments for classification. It is probably not useful here. Can be remove.